### ISSUE TITLE
Incorrect warning: "WARNING: Project targetting '>= 0.42.0' but tried to use feature introduced in '0.47.0': build_always_stale arg in custom_target"

### ISSUE THREAD

**Issue by mcatanzaro** [mesonbuild/meson#4110]
I have a project (Epiphany) which is getting this warning:

```
WARNING: Project targetting '>= 0.42.0' but tried to use feature introduced in '0.47.0': build_always_stale arg in custom_target
```

However, we don't use `build_always_stale_arg` anywhere in Epiphany (or libdazzle, its only subproject), so the warning is wrong.

This is probably somehow related to #4107.

**Comment by nirbheek** [comment]
Duplicate of https://github.com/mesonbuild/meson/issues/3904

**Comment by nirbheek** [comment]
This was fixed in 0.47.2
